db/crud.go

```diff
@@ -990,6 +990,13 @@ func (db *Database) updateAndReturnDoc(
 			doc.History[newRevID].Channels = channelSet
 		}
 
+		//Need to check and add attachments here to ensure the attachment is within size constraints
+		attachmentErr := db.setAttachments(newAttachments)
+		if attachmentErr != nil{
+			err = errors.Wrap(attachmentErr, "Error adding attachment")
```

Comment:
Could we create a HTTP error instead like the errors further up? That way we can set a more appropriate status code than 500 (like HTTP 413 Request Entity Too Large)